erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -656,17 +655,16 @@ def get_item_details(self, args=None, for_update=False):
 			'has_batch_no'			: item.has_batch_no,
 			'sample_quantity'		: item.sample_quantity
 		})
-		for d in [["Account", "expense_account", "default_expense_account"],
-			["Cost Center", "cost_center", "cost_center"]]:
-				company = frappe.db.get_value(d[0], ret.get(d[1]), "company")
-				if not ret[d[1]] or (company and self.company != company):
-					ret[d[1]] = frappe.get_cached_value('Company',  self.company,  d[2]) if d[2] else None
 
 		# update uom
 		if args.get("uom") and for_update:
 			ret.update(get_uom_details(args.get('item_code'), args.get('uom'), args.get('qty')))
 
-		if not ret["expense_account"]:
+		if self.purpose == 'Material Issue':
+			ret["expense_account"] = (item.get("expense_account") or
```

Comment:
If expense account is not set in Item master, then it should look for Item Group and then Company record.

Reply:
Done